modules/store/src/models.ts

```diff
@@ -80,6 +95,8 @@ export type Creator<
   R extends object = object
 > = FunctionWithParametersType<P, R>;
 
+type Primitive = string | number | bigint | boolean | symbol | null | undefined;
```

Comment:
I wonder if there's a reason Typescript doesn't provide this type already...

Reply:
Their site says

> object is a type that represents the non-primitive type, i.e. anything that is not number, string, boolean, bigint, symbol, null, or undefined.

So it seems like checking if something `extends` object would be sufficient.

Perhaps I shouldn't have an explicit check for primitives then.

Reply:
I need an explicit check because `unknown` does not extend `object`, and so it must be differentiated (but we can't check `extends unknown` because every type is assignable to `unknown`).